mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -171,117 +190,207 @@ public void commitJob(JobContext originalContext) throws IOException {
+    JobConf jobConf = jobContext.getJobConf();
 
-    // Remove the job's temp directory recursively.
-    // Intentionally used foreach on a single item. Using the Tasks API here only for the retry capability.
-    Tasks.foreach(location)
-        .retry(3)
-        .suppressFailureWhenFinished()
-        .onFailure((file, exc) -> LOG.debug("Failed on to remove directory {} on cleanup job", file, exc))
-        .run(file -> {
-          Path toDelete = new Path(file);
-          FileSystem fs = Util.getFs(toDelete, jobContext.getJobConf());
-          fs.delete(toDelete, true);
-        }, IOException.class);
+    LOG.info("Cleaning for job {} started", jobContext.getJobID());
+    Map<String, String> outputs = SerializationUtil.deserializeFromBase64(jobConf.get(InputFormatConfig.OUTPUT_TABLES));
+
+    for (String table : outputs.keySet()) {
+      String tableLocation = generateJobLocation(outputs.get(table), jobConf, jobContext.getJobID());
+      LOG.info("Cleaning table {} on location: {}", table, tableLocation);
+
+      // Remove the job's temp directory recursively.
+      // Intentionally used foreach on a single item. Using the Tasks API here only for the retry capability.
+      Tasks.foreach(tableLocation)
+          .retry(3)
+          .suppressFailureWhenFinished()
+          .onFailure((file, exc) -> LOG.debug("Failed on to remove directory {} on cleanup job", file, exc))
+          .run(file -> {
+            Path toDelete = new Path(file);
+            FileSystem fs = Util.getFs(toDelete, jobContext.getJobConf());
+            fs.delete(toDelete, true);
+          }, IOException.class);
+    }
+
+    LOG.info("Cleaning for job {} finished", jobContext.getJobID());
+  }
+
+  /**
+   * Executor service for parallel handling of file reads. Should be shared when committing multiple tables.
+   * @param conf The configuration containing the pool size
+   * @return The generated executor service
+   */
+  private static ExecutorService fileReaderExecutor(Configuration conf) {
+    int size = conf.getInt(InputFormatConfig.COMMIT_FILE_THREAD_POOL_SIZE,
+        InputFormatConfig.COMMIT_FILE_THREAD_POOL_SIZE_DEFAULT);
+    return Executors.newFixedThreadPool(
+        size,
+        new ThreadFactoryBuilder()
+            .setDaemon(true)
+            .setPriority(Thread.NORM_PRIORITY)
+            .setNameFormat("iceberg-commit-file-pool-%d")
+            .build());
+  }
+
+  /**
+   * Executor service for parallel handling of table manipulation. Could return null, if no parallelism is possible.
+   * @param conf The configuration containing the pool size
+   * @return The generated executor service, or null if executor is not needed.
+   */
+  private static ExecutorService tableExecutor(Configuration conf, int outputSize) {
+    int size = conf.getInt(InputFormatConfig.COMMIT_TABLE_THREAD_POOL_SIZE,
+        InputFormatConfig.COMMIT_TABLE_THREAD_POOL_SIZE_DEFAULT);
+    size = Math.max(outputSize, size);
```

Comment:
Would it make sense to use `Math.min` instead? If the outputSize is smaller than the conf, we'll end up launching threads unnecessarily. If outputSize is larger, using `min` would still make sense as it would enforce the conf size setting provided by the user (maybe only if they provided one explicitly, other than the default)?

Reply:
Yeah, that was the plan 😄